torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -134,88 +134,105 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 }
 
 // See NOTE [ Autograd View Variables ] for details.
-inline Tensor as_view(const Tensor & base, const Tensor& tensor, bool is_differentiable,
-        c10::optional<std::function<Tensor(const Tensor&)>> view_func=c10::nullopt,
-        CreationMeta creation_meta=CreationMeta::DEFAULT) {
-  auto base_var = Variable(base);
-  if (base_var.is_view()) {
-    // Set `view_func` using the root base as input.
-    // `view_func` is used to recover views in backward when either as_strided is not supported
-    // or the view function changes the metadata which is not recorded by as_strided
-    // See Note [View + Inplace update on base tensor] and [View + Inplace update on view tensor]
-    // for more details how we use this function in backward.
-    auto diff_view_meta = static_cast<DifferentiableViewMeta*>(torch::autograd::impl::get_autograd_meta(base_var));
-    if (view_func.has_value()) {
-      auto fn = view_func.value();
-      // both current_view and it's parent have a view_func
-      if (diff_view_meta->has_view_fn()) {
-        auto prev_fn = diff_view_meta->view_fn();
-        view_func = [=](const at::Tensor& root_base) {
-          auto temp = prev_fn(root_base);
-          return fn(temp);
-        };
+inline Tensor as_view(const Tensor & base, const Tensor & tensor, bool is_bw_differentiable,
+        bool is_fw_differentiable, c10::optional<std::function<Tensor(const Tensor&)>> view_func=c10::nullopt,
+        CreationMeta creation_meta=CreationMeta::DEFAULT, bool allow_tensor_metadata_change=true) {
+  if (!isForwardADEnabled()) {
```

Comment:
I'm not so hot about the code duplication here (although it's not the worst thing in the world). Apparently, this is because `c10::optional<ViewInfo>` default construction is SOOOO expensive it doubles the overhead on the benchmark (by the way, that's weird, and worth looking into--I still kind of feel like std::function is to blame). But if the default construction is the problem, it should still be possible to reuse the `is_bw_differentiable` code below (but don't allocate the temp yet), and then do an early return. IDK, it's not worth blocking the PR on, just keep this in mind for next time.